code/modules/mob/living/carbon/superior_animal/robots/subtype/mining/mining_drone.dm

```diff
@@ -0,0 +1,130 @@
+/mob/living/carbon/superior_animal/robot/mining
+	name = "Generic Mining Drone"
+	desc = "A small, unbranded drone, it has a drill and seem to search for shiny stuff."
+	gender = NEUTER
+	icon = 'icons/mob/battle_roomba.dmi'
+	icon_state = "mining_drone"
```

Comment:
death icon?

Reply:
It shouls explode on death. 


...And I just realized I forgot to test that part...

Reply:
I just tested it, the minebot get gibbed on death. So it doesn't need a death icon.